### PR TITLE
[test](inverted index)Add fault injection cases for index writing

### PR DESCRIPTION
**Pull request by qidaye**
## Proposed changes

Issue Number: close #xxx

### CODE REVIEW

**Comment by qidaye** [comment]
run buildall

**Comment by qidaye** [comment]
run buildall

**Comment by qidaye** [comment]
run build'all

**Comment by qidaye** [comment]
run buildall

**Comment by qidaye** [comment]
run buildall

**Review by hello-stephen — DISMISSED** [review]

File: be/src/olap/rowset/segment_v2/inverted_index_file_writer.cpp [hunk 1]
```diff
@@ -337,110 +357,125 @@ int64_t InvertedIndexFileWriter::write_v1() {
             auto* new_index_info = _file_info.add_index_info();
             *new_index_info = index_info;
         } catch (CLuceneError& err) {
-            LOG(ERROR) << "CLuceneError occur when close idx file "
-                       << InvertedIndexDescriptor::get_index_file_path_v1(_index_path_prefix,
-                                                                          index_id, index_suffix)
+            auto index_path = InvertedIndexDescriptor::get_index_file_path_v1(
+                    _index_path_prefix, index_id, index_suffix);
+            LOG(ERROR) << "CLuceneError occur when close idx file " << index_path
```
Comment (airborne12) [inline be/src/olap/rowset/segment_v2/inverted_index_file_writer.cpp:362]: CLuceneError occur when write_v1 idx file

File: be/src/olap/rowset/segment_v2/inverted_index_file_writer.cpp [hunk 2]
```diff
@@ -337,110 +357,125 @@ int64_t InvertedIndexFileWriter::write_v1() {
...
                        << " error msg: " << err.what();
 
-            throw err;
+            return Status::Error<ErrorCode::INVERTED_INDEX_CLUCENE_ERROR>(
+                    "CLuceneError occur when close idx file: {}, error msg: {}", index_path,
```
Comment (airborne12) [inline be/src/olap/rowset/segment_v2/inverted_index_file_writer.cpp:366]: same as upper comment.

File: be/src/olap/rowset/segment_v2/inverted_index_fs_directory.cpp [hunk 3]
```diff
@@ -353,6 +370,7 @@ void DorisFSDirectory::FSIndexOutput::close() {
         }
     } else {
         LOG(WARNING) << "File writer is nullptr, ignore finalize and close.";
+        _CLTHROWA(CL_ERR_IO, "close file writer error, _writer = nullptr");
```
Comment (airborne12) [inline be/src/olap/rowset/segment_v2/inverted_index_fs_directory.cpp:373]: why throw error here?

Comment (qidaye) [reply be/src/olap/rowset/segment_v2/inverted_index_fs_directory.cpp:373]: If just print the warn log here, the dat file is written successfully, but the idx is not.

File: be/src/olap/rowset/segment_v2/inverted_index_file_writer.cpp (hunk 2 above)
Comment (qidaye) [reply be/src/olap/rowset/segment_v2/inverted_index_file_writer.cpp]: done

File: be/src/olap/rowset/segment_v2/inverted_index_file_writer.cpp (hunk 1 above)
Comment (qidaye) [reply be/src/olap/rowset/segment_v2/inverted_index_file_writer.cpp]: done

**Comment by qidaye** [comment]
run buildall

**Comment by qidaye** [comment]
run buildall

**Comment by qidaye** [comment]
run buildall

**Comment by qidaye** [comment]
run buildall

**Comment by qidaye** [comment]
run buildall

**Review by airborne12 — APPROVED** [review]
LGTM

**Review by hello-stephen — APPROVED** [review]

**Pull request merged**